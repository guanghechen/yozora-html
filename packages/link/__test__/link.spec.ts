import type { Link as ILink } from '@yozora/ast'
import { renderChildren } from '../../../jest.setup'
import renderLink from '../src'

describe('snapshot', function () {
  it('basic', function () {
    const node = {
      type: 'link',
      url: 'https://github.com/guanghechen/yozora',
      title: 'yozora',
      children: [
        {
          type: 'text',
          value: 'yozora is cool!',
        },
      ],
    }
    expect(renderLink(node as ILink, renderChildren)).toMatchSnapshot()
  })

  it('basic-2', function () {
    const node = {
      type: 'link',
      url: 'https://github.com/yozorajs/yozora-html',
      title: 'yozora-html',
      children: [
        {
          type: 'emphasis',
          children: [
            {
              type: 'text',
              value: 'Hello',
            },
          ],
        },
        {
          type: 'text',
          value: ', ',
        },
        {
          type: 'strong',
          children: [
            {
              type: 'text',
              value: 'World',
            },
          ],
        },
      ],
    }
    expect(renderLink(node as ILink, renderChildren)).toMatchSnapshot()
  })
})
