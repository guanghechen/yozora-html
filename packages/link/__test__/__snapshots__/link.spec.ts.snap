// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot basic 1`] = `"<a class="yozora-link" href="https://github.com/guanghechen/yozora" title="yozora" target="_blank" rel="noopener,noreferrer"><span class="yozora-text">yozora is cool!</span></a>"`;

exports[`snapshot basic-2 1`] = `"<a class="yozora-link" href="https://github.com/yozorajs/yozora-html" title="yozora-html" target="_blank" rel="noopener,noreferrer"><em class="yozora-emphasis"><span class="yozora-text">Hello</span></em><span class="yozora-text">, </span><strong class="yozora-strong"><span class="yozora-text">World</span></strong></a>"`;
